### PR TITLE
Changed tfy buildkit helm chart and converted it to StatefulSet

### PR DESCRIPTION
**Pull request by mitanshudodia**

### CODE REVIEW

File: charts/tfy-buildkitd-service/templates/deployment.yaml [hunk 1]
```diff
@@ -76,3 +85,13 @@ spec:
...
         {{- toYaml . | nindent 8 }}
       {{- end }}
+  volumeClaimTemplates:
+    - metadata:
+        name: buildkit-storage
+      spec:
+        accessModes: [ "ReadWriteOnce" ]
+        resources:
+          requests:
+            storage: 100Gi
```
Comment (debajyoti-truefoundry) [inline charts/tfy-buildkitd-service/templates/deployment.yaml:95]: Expose in values.

File: charts/tfy-buildkitd-service/templates/configmaps.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: buildkit-config
+  labels:
+    {{- include "buildkitd-service.labels" . | nindent 4 }}
+data:
+  buildkitd.toml: |
```
Comment (debajyoti-truefoundry) [inline charts/tfy-buildkitd-service/templates/configmaps.yaml:8]: The content for this should be present in the values file.

File: charts/tfy-buildkitd-service/values.yaml [hunk 3]
```diff
@@ -15,7 +15,16 @@ image:
   pullPolicy: IfNotPresent
```
Comment (debajyoti-truefoundry) [inline charts/tfy-buildkitd-service/values.yaml:15]: Can we please remove autoscaling related values?

File: charts/tfy-buildkitd-service/templates/deployment.yaml [hunk 4]
```diff
@@ -76,3 +85,13 @@ spec:
       tolerations:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+  volumeClaimTemplates:
+    - metadata:
+        name: buildkit-storage
+      spec:
+        accessModes: [ "ReadWriteOnce" ]
```
Comment (debajyoti-truefoundry) [inline charts/tfy-buildkitd-service/templates/deployment.yaml:92]: Move this to values.

File: charts/tfy-buildkitd-service/values.yaml [hunk 5]
```diff
@@ -15,7 +15,44 @@ image:
...
+
+  ## @param storage.accessModes Access mode for tfyBuildkitdService
+  accessModes: 
+    - ReadWriteOnce
+
+  ## @param storage.storageClassName Storage class name for tfyBuildkitdService
+  storageClassName: ""
+
+  ## @param storage.size Size of the storage for tfyBuildkitdService
+  size: 100Gi
```
Comment (debajyoti-truefoundry) [inline charts/tfy-buildkitd-service/values.yaml:55]:
```suggestion
  size: 200Gi
```

**Review by akashg3627 — APPROVED** [review]

File: charts/tfy-buildkitd-service/templates/deployment.yaml
Comment (akashg3627) [inline charts/tfy-buildkitd-service/templates/deployment.yaml:1]: change the filename

File: charts/tfy-buildkitd-service/values.yaml [hunk 6]
```diff
@@ -15,7 +15,41 @@ image:
...
+  all = true
+
+## Storage configuration for tfyBuildkitdService
+##
+storage:
+  ## @param storage.accessModes Access mode for tfyBuildkitdService
+  accessModes: 
+    - ReadWriteOnce
+  ## @param storage.storageClassName Storage class name for tfyBuildkitdService
+  storageClassName: ""
```
Comment (akashg3627) [inline charts/tfy-buildkitd-service/values.yaml:50]: should be put some default here?

Comment (akashg3627) [reply charts/tfy-buildkitd-service/values.yaml:50]: ignore if it will pick cluster default

**Pull request merged**